metrics/internal/collectors/registry.go

```diff
@@ -49,12 +49,13 @@ func RegisterCustomResourceCollectors(registry *prometheus.Registry, opts *optio
 }
 
 var pvStoreEnabled bool
-var pvStore = internalcache.NewPersistentVolumeStore()
+var pvStore *internalcache.PersistentVolumeStore
 
 func enablePVStore(opts *options.Options) {
+	pvStore = internalcache.NewPersistentVolumeStore(opts)
 	client := clientset.NewForConfigOrDie(opts.Kubeconfig)
 	lw := internalcache.CreatePersistentVolumeListWatch(client, "")
-	reflector := cache.NewReflector(lw, &corev1.PersistentVolume{}, pvStore, 0)
+	reflector := cache.NewReflector(lw, &corev1.PersistentVolume{}, pvStore, 30*time.Second)
```

Comment:
There could be a lot of PVs. We shouldn't resync too aggressively. Maybe 2 min is better.